src/go/intelligence/utils/utils.go

```diff
@@ -201,16 +239,95 @@ func findDashboards() models.AllDashboardsFetched {
 	return data
 }
 
+//GetSilences - function for get request on /api/v1/silences alertmanager endpoint for fetching all silences.
+func GetSilences() (models.AllSilences, int, int, int, error) {
+	var data models.AllSilences
+	apiurl := ValidateURL(ConfigJSON.Server.CMSMONURL, ConfigJSON.Server.GetSilencesAPI) //GET API for fetching all AM Silences.
+
+	req, err := http.NewRequest("GET", apiurl, nil)
+	if err != nil {
+		log.Printf("Request Error, error: %v\n", err)
+		return data, 0, 0, 0, err
+	}
+	req.Header.Add("Accept-Encoding", "identity")
+	req.Header.Add("Accept", "application/json")
+
+	timeout := time.Duration(ConfigJSON.Server.HTTPTimeout) * time.Second
+	client := &http.Client{Timeout: timeout}
+
+	if ConfigJSON.Server.Verbose > 1 {
+		log.Println("GET", apiurl)
+	} else if ConfigJSON.Server.Verbose > 1 {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("Request: ", string(dump))
+		}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Printf("Response Error, error: %v\n", err)
+		return data, 0, 0, 0, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		log.Printf("Http Response Code Error, status code: %d", resp.StatusCode)
+		return data, 0, 0, 0, errors.New("Respose Error")
+	}
+
+	defer resp.Body.Close()
+
+	byteValue, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		log.Printf("Unable to read JSON Data from AlertManager Silence GET API, error: %v\n", err)
+		return data, 0, 0, 0, err
+	}
+
+	if ConfigJSON.Server.Verbose > 1 {
+		dump, err := httputil.DumpResponse(resp, true)
+		if err == nil {
+			log.Println("Response: ", string(dump))
+		}
+	}
+
+	err = json.Unmarshal(byteValue, &data)
+	if err != nil {
+		if ConfigJSON.Server.Verbose > 0 {
+			log.Println(string(byteValue))
+		}
+		log.Printf("Unable to parse JSON Data from AlertManager Silence GET API, error: %v\n", err)
+		return data, 0, 0, 0, err
+	}
+
+	expiredLength := 0
```

Comment:
I don't really see needs for making new variables and complicating function signature. Why not to put all counters into data structure itself?

Reply:
Fixed this issue.